packages/veritone-react-common/src/components/LogoDetectionEngineOutput/index.js

```diff
@@ -0,0 +1,198 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import classNames from 'classnames';
+import IconButton from 'material-ui/IconButton';
+import ZoomOutMap from 'material-ui-icons/ZoomOutMap';
+import Select from 'material-ui/Select';
+import { MenuItem } from 'material-ui/Menu';
+import EngineOutputHeader from '../EngineOutputHeader';
+import PillButton from '../Parts/Buttons/PillButton';
+import { msToReadableString } from '../../helpers/time'
+import styles from './styles.scss';
+
+export default class LogoDetectionEngineOutput extends Component {
+	static propTypes = {
+		label: PropTypes.string,
+		data: PropTypes.arrayOf(Object),
+		mediaPlayerTime: PropTypes.number,
+		className: PropTypes.string,
+		headerClassName: PropTypes.string,
+		bodyClassName: PropTypes.string,
+		itemClassName: PropTypes.string,
+		itemLabelClassName: PropTypes.string,
+		itemInfoClassName: PropTypes.string,
+		onItemSelected: PropTypes.func,
+		onEngineSelected: PropTypes.func,
+		onExpandViewClicked: PropTypes.func
+	};
+
+	static defaultProps = {
+		label: 'Logo Recognition',
+		data: [],
+		mediaPlayerTime: -1
+	};
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			isExpanded: false
+		};
+
+		this.onEngineChanged = this.onEngineChanged.bind(this);
+		this.onExpandViewClicked = this.onExpandViewClicked.bind(this);
+	}
+
+	//================================================================================
+	//=============================Interaction Functions==============================
+	//================================================================================
+	onItemSelected = item => (event) => {
```

Comment:
this technique (currying) is a clever way around the jsx-no-bind eslint rule, but it has the same negative effect that we try to avoid with that rule -- basically this function returns a new function instance every time it's called, react sees that the new instance isn't `===` to the last one it saw, so it updates the DOM. We try to avoid doing that as much as possible for performance reasons (it causes unnecessary DOM updates and GC runs).

> A bind call or arrow function in a JSX prop will create a brand new function on every single render. This is bad for performance, as it will result in the garbage collector being invoked way more than is necessary. It may also cause unnecessary re-renders if a brand new function is passed as a prop to a component that uses reference equality check on the prop to determine if it should update.

*In this case, IMO we can leave it as-is*, because my experience tells me the performance hit won't be worth the refactor in this case. But if we were to change it, the correct way is typically to wrap the argument passing logic in the child component. So rather than this:

```javascript
<PillButton onClick={ this.onItemSelected(item) } />
```

you would do

```javascript
<PillButton 
  onClick={ this.onItemSelected }  // always sees the same function instance
  item={ item }
/>

// then in PillButton.js

handleClick = () => {
  // logic of what to pass to the callback goes here instead
  this.props.onClick(this.props.item)
}
```

(you can see that this isn't always optimal for making the code understandable. So we make considerations based on whether the performance hit will matter. but just FYI this is a pretty important concept.)